alchemist/alchemist-euclidean-geometry/src/main/kotlin/it/unibo/alchemist/model/implementations/geometry/euclidean2d/SlopeInterceptLine2D.kt

```diff
@@ -0,0 +1,161 @@
+                    val slope = toVector.run { y / x }
+                    SlopeInterceptLine2D(slope, first.y - slope * first.x, first::newFrom)
+                }
+            }
+        }
+
+        /**
+         * Solves the quadratic equation [a] * x^2 + [b] * x + [c] = 0.
+         */
+        private fun solveQuadraticEquation(a: Double, b: Double, c: Double): List<Double> {
+            val determinant = b.pow(2) - 4 * a * c
+            return when {
+                determinant < 0.0 -> emptyList()
+                MathUtils.fuzzyEquals(determinant, 0.0) -> listOf(-b)
+                else -> listOf(-b + sqrt(determinant), -b - sqrt(determinant))
+            }.map { it / (2 * a) }
+        }
+    }
+
+    override val isHorizontal: Boolean = MathUtils.fuzzyEquals(slope, 0.0)
+
+    override val isVertical: Boolean = slope.isNaN()
+
+    override fun contains(point: P): Boolean = when {
+        isVertical -> MathUtils.fuzzyEquals(xIntercept, point.x)
+        else -> MathUtils.fuzzyEquals(findY(point.x), point.y)
+    }
+
+    /**
+     * Solves the line equation for the given [x]. Throws an [UnsupportedOperationException] if the line [isVertical].
+     */
+    fun findY(x: Double): Double = when {
+        isVertical -> throw UnsupportedOperationException("can't compute y coordinate for vertical line")
+        else -> slope * x + yIntercept
+    }
+
+    override fun findPoint(x: Double): P = createPoint(x, findY(x))
+
+    override fun isParallelTo(other: Line2D<*>): Boolean =
+        isVertical && other.isVertical || MathUtils.fuzzyEquals(slope, other.slope)
+
+    /**
+     * Checks if two lines coincide.
+     */
+    fun coincidesWith(other: Line2D<*>): Boolean = when {
+        !isParallelTo(other) -> false
+        isVertical -> MathUtils.fuzzyEquals(xIntercept, other.xIntercept)
+        else -> MathUtils.fuzzyEquals(yIntercept, other.yIntercept)
+    }
+
+    override fun intersect(other: Line2D<P>): Intersection2D<P> = when {
+        coincidesWith(other) -> Intersection2D.InfinitePoints
+        /*
+         * But not coincident.
+         */
+        isParallelTo(other) -> Intersection2D.None
+        else -> Intersection2D.SinglePoint(when {
+            /*
+             * other can't be vertical as lines are not parallel at this point.
+             */
+            isVertical -> other.findPoint(xIntercept)
+            other.isVertical -> findPoint(other.xIntercept)
+            else -> findPoint((other.yIntercept - yIntercept) / (slope - other.slope))
+        })
+    }
+
+    /**
+     * Intersects a line and a circle. Radius must be positive.
+     * Intersection is performed by plugging the line equation in the circle equation and solving the resulting
+     * quadratic equation.
+     * Circle equation: (x - [center].x)^2 + (y - [center].y)^2 = r^2.
+     * Line equation: y = [slope] * x + [yIntercept] unless [isVertical], x = [xIntercept] otherwise.
+     */
+    @Suppress("MagicNumber")
```

Comment:
2, what should I do? It's plenty of two when solving a quadratic equation

Reply:
That code combines a circle equation and a line equation and solves the resulting quadratic equation, it's math rather than Kotlin

Reply:
Sure it's 2? 2 should not be a magic number already. Something smells here. If it's 2, we need to make an exception for it. If it's something else, we need to understand what it is

Reply:
Aaah it's -2. I'll move the minus to another element of the product, so that Detekt doesn't complain.

Reply:
Not so fast, let's discuss. Should -2 be a magic number? I think we might consider suppressing it... What do you think?

Reply:
Ignored numbers now are: `-1, 0, 1, 2`. I can't see a valid reason for ignoring `-1` and `2` but not `-2`. So yes, I think it should be suppressed.
Additionally, since `2` is ignored, it's plenty of workarounds to use `-2`, as I did a while ago: `-2 * x` can be written as `2 * -x`, or `x - 2` as `-(2 - x)`.

In general, if a number is ignored, I don't see a valid point for not ignoring its opposite.

Reply:
The workaround is *way* worse than the suppression. -2 should not be a magic number.

Reply:
Ok, I added it to the ignored numbers and removed the workaround. By the way, I was wrong, ignored numbers are: `-2, -1, 0, 0.5, 1, 2, 3, 4, 10, 360.0`.